main.cpp

```diff
@@ -1,81 +1,127 @@
+#include <sys/stat.h>
 #include <iostream>
-
-#include <character.h>
+#include <fstream>
+#include "character.h"
 #include <string>
+#include <vector>
 
 void battle(Character& character1, Character& character2){
   while(character1.IsAlive() && character2.IsAlive()){
-    std::cout << character1;
-    std::cout << character2;
-    std::cout << character1.GetName() << " -> " << character2.GetName() << '\n';
-
     character1.Attack(character2);
-    std::cout << character1;
-    std::cout << character2;
-
     if(character2.IsAlive()){
-      std::cout << character2.GetName() << " -> " << character1.GetName() << '\n';
       character2.Attack(character1);
     }
   }
 
   if(character1.IsAlive()){
-    std::cout << character2.GetName() << " died. " << character1.GetName() << " wins.\n";
+    std::cout << character1.GetName() << " wins. Remaining HP:" << character1.GetHealth() << '\n';
   }
   else{
-    std::cout << character1.GetName() << " died. " << character2.GetName() << " wins.\n";
+    std::cout << character2.GetName() << " wins. Remaining HP:" << character2.GetHealth() << '\n';
   }
 }
 
-int checkHealthOrDamageInputArgument(char *argument){
-  int number;
-  try {
-    number = std::atoi(argument);
-  } catch (...) {}
+std::vector<std::string> splittedString(std::string text, char delimiter){
+  std::vector<std::string> splittedStrings;
+  std::string addString = "";
 
-  return number;
-}
-
-int main(int argc, char *argv[])
-{
-  if(argc != 7){
-    std::cout << "Something is wrong with the arguments.\nExample: Maple 150 10 Sally 45 30\n";
-    return 1;
-  }
-  else{
-    std::string character1_name(argv[1]);
-
-    int character1_health = checkHealthOrDamageInputArgument(argv[2]);
-    if(character1_health == 0){
-      std::cout << argv[2] << " is not valid for health\n";
-      return 1;
+  for(unsigned int i = 0; i < text.length(); i++){
+    if(text[i] == delimiter){
+      splittedStrings.push_back((addString));
+      addString = "";
     }
-
-    int character1_damage = checkHealthOrDamageInputArgument(argv[3]);
-    if(character1_damage == 0){
-      std::cout << argv[3] << " is not valid for damage\n";
-      return 1;
+    else{
+      addString += text[i];
     }
+  }
 
-    Character character1(character1_name, character1_health, character1_damage);
+  splittedStrings.push_back((addString));
 
-    std::string character2_name(argv[4]);
+  return splittedStrings;
+}
 
-    int character2_health = checkHealthOrDamageInputArgument(argv[5]);
-    if(character2_health == 0){
-      std::cout << argv[5] << " is not valid for health\n";
-      return 1;
-    }
+bool fileExists(const std::string& fileName)
+{
+	std::ifstream f(fileName.c_str());
```

Comment:
ilyenkor nem illene close-olni is?  erre azert nem csinalnek egy kulon fuggvenyt.

Reply:
Kiszedtük a fileExists függvényt.